### PR TITLE
user.py: replace user_mod with ldap.update_entry()

### PR DESCRIPTION
**Pull request by Tiboris**
Refactoring user_add class to use 'ldap.update_entry()' call
    instead of api call 'user_mod' when --noprivate option is used.

    https://pagure.io/freeipa/issue/5788

### CODE REVIEW

File: ipaserver/plugins/user.py [hunk 1]
```diff
@@ -603,11 +603,14 @@ def post_callback(self, ldap, dn, entry_attrs, *keys, **options):
...
                 entry_attrs.update(desc_attr)
+                entry_attrs._orig['description'] = list(
+                    desc_attr['description'])  # The entry_attrs.update()
+                # does not change the entry_attrs._orig
             if 'description' in entry_attrs and NO_UPG_MAGIC in entry_attrs['description']:
                 entry_attrs['description'].remove(NO_UPG_MAGIC)
-                kw = {'setattr': unicode('description=%s' % ','.join(entry_attrs['description']))}
                 try:
-                    self.api.Command['user_mod'](keys[-1], **kw)
+                    self._exc_wrapper(
```
Comment (MartinBasti) [inline ipaserver/plugins/user.py]: is really needed to call _exc_wrapper?  ldap.update_entry(entry) should return the same exceptions. IMO it is too complex with _exc_wrapper.

File: ipaserver/plugins/user.py [hunk 2]
```diff
@@ -603,11 +603,14 @@ def post_callback(self, ldap, dn, entry_attrs, *keys, **options):
             if not options.get('all', False):
                 desc_attr = ldap.get_entry(dn, ['description'])
                 entry_attrs.update(desc_attr)
+                entry_attrs._orig['description'] = list(
```
Comment (MartinBasti) [inline ipaserver/plugins/user.py]: How about to move this whole block 20 lines below, where `newentry = ldap.get_entry(dn, ['*'])` is. IIUC that call will download the whole user_entry, so you can just delete `NO_UPG_MAGIC` if needed and you even save one more LDAP operation without ugly hacking of LDAPEntry.

**Comment by stlaz** [comment]
+ please don't indent the commit message.

**Comment by MartinBasti** [comment]
master:

* d73ec06cb3c8042f90d1520485aa9b0f2e7c8fbe user.py: replace user_mod with ldap.update_entry()


ipa-4-5:

* daeac31ba54d88159b46d61fc443ccd45902f8f2 user.py: replace user_mod with ldap.update_entry()

**Pull request closed without merging**